### ISSUE TITLE
Society B356 Kutchin is coded with invalid code 0 for variable B004

### ISSUE THREAD

**Issue by xrotwang** [D-PLACE/dplace-data#187]
See https://github.com/D-PLACE/dplace-data/blob/8ca9af9984d23c7dd01cddd605ed3e57d57dfbdf/datasets/Binford/data.csv#L11130
and the allowed codes at https://github.com/D-PLACE/dplace-data/blob/master/datasets/Binford/codes.csv#L5-L8

**Comment by xrotwang** [comment]
We should catch these cases with pydplace

**Comment by SimonGreenhill** [comment]
Good idea, let's use these to validate. God knows how many more typos and glitches there are ... :/

**Comment by xrotwang** [comment]
Funny enough that seems to be the only error of this type:
```bash
$ dplace --repos=. check
ERROR   undefined code: B004:0
```

**Comment by xrotwang** [comment]
@SimonGreenhill @kirbykat unfortunately, I found a couple more data consistency problems. For the above check, I only looked at variables classified as `Categorical`. But in SCCS we have quite a few variables classified as `Ordinal`, but still with defined, batched codes, e.g. [here](https://github.com/D-PLACE/dplace-data/blob/39a5a48465cc8dc341646ae71f00a7677c5a1edf/datasets/SCCS/codes.csv#L5041-L5046)
```
SCCS811,NA,Missing data,NA
SCCS811,1,1,1
SCCS811,2,2,2
SCCS811,3,3,3
SCCS811,4,4,4
SCCS811,5,5 or more,5 or more
```
In this case, there are three societies coded with `0`.

This raises the question, whether it should be considered an error if an ordinal variable lists distinct codes in `codes.csv` - i.e. we would disallow classifying variables which batch values together as `ordinal`? But we could also adopt the convention that ordinal variables can batch values, but then have to provide codes for **all** datapoints in `codes.csv`.

**Comment by xrotwang** [comment]
I guess, right now, the semantics is more or less as follows:
- `Categorical`: All possible codes are listed in `codes.csv`. Icons should be chosen from a categorical scale.
- `Ordinal`: All actually measured codes are listed in `codes.csv` and have code IDs which can be ordered as numbers. Icons should be chosen from an ordinal scale. Note: This allows for batching!
- `Continuous`: Values are not listed in `codes.csv`.

If we stick to this logic, we have to
- add a code `0` for variable `SCCS811`
- fix datapoint for variable `SCCS1926` and society `SCCS140`: The current value is `54`, which does not have a code
- classify `SCCS1862` and `SCCS1756` as continuous.

**Comment by SimonGreenhill** [comment]
Hmm. This makes sense to me, thanks for pinning it down. Do we know if states 0 and 54 in those variables are typos or in the originals?

**Comment by xrotwang** [comment]
The `0` makes some sense, so I'd say it's a missing code. The `54` otoh doesn't make sense for

> on a scale of 0-10

**Comment by kirbykat** [comment]
Hi, thanks for catching this, @xrotwang. The logic you lay out for what is/is not listed in codes.csv sounds good to me. Let me upload the still-to-be-finally-cleaned version of the SCCS that Xinyu and Sreeja have worked on - in re-checking variable names and code definitions, they sometimes caught individual codes that should have been in our codes.csv file, but were not.

**Comment by xrotwang** [comment]
@kirbykat can we resolve some of these issues?

**Comment by kirbykat** [comment]
>Re: Society B356 Kutchin is coded with invalid code 0 for variable B004:

The code should be "1" (most important subsistence activity is hunting).

**Comment by kirbykat** [comment]
@xrotwang I'll get to the SCCS problems in the next days. I'm hoping they've been resolved in the hiwi code/variable correction effort.

**Comment by xrotwang** [comment]
Original issue resolved, remainder moved to #231
